core/src/verify.rs

```diff
@@ -1233,9 +1243,53 @@ mod test {
     #[test]
     /// Test the case where the transaction commit is invalid because it is extra-agenda transaction phase.
     /// This test case is ignored because the extra-agenda transaction is not implemented yet.
-    // TODO: enable this test case when the extra-agenda transaction is implemented.
     fn phase_mismatch_for_transaction_commit3() {
-        todo!("Implement this test");
+        let (validator_keypair, reserved_state, mut csv) = setup_test(4);
+        // Apply agenda commit
+        let agenda_transactions_hash = calculate_agenda_transactions_hash(csv.phase.clone());
+        let agenda: Agenda = Agenda {
+            author: reserved_state.query_name(&validator_keypair[0].0).unwrap(),
+            timestamp: 1,
+            transactions_hash: agenda_transactions_hash,
+            height: csv.header.height + 1,
+            previous_block_hash: csv.header.to_hash256(),
+        };
+
+        // delegator: member-0, delegatee: member-1
+        let delegator = reserved_state.members[0].clone();
+        let delegator_private_key = validator_keypair[0].1.clone();
+        let delegatee = reserved_state.members[1].clone();
+
```

Comment:
Remove this empty line please since it seems better to keep codes to generate DelegationTransactionData.